db/attachment.go

```diff
@@ -107,7 +107,11 @@ func (db *Database) storeAttachments(doc *document, body Body, generation int, p
 			}
 			// Try to look up the attachment in ancestor attachments
 			if parentAttachments == nil {
-				parentAttachments = db.retrieveAncestorAttachments(doc, parentRev, docHistory)
+				var err error
+				parentAttachments, err = db.retrieveAncestorAttachments(doc, parentRev, docHistory)
```

Comment:
It seems like the crux of this regression is that parentAttachments should be nil here (so that it falls through to the "missing digest" error below).  In the failing test (pre-fix), what's actually being returned here?

Reply:
Pre-fix the parent attachments appears to pick up the attachment which it shouldn't have:

![image](https://user-images.githubusercontent.com/3264971/52371047-644a9000-2a4c-11e9-9a21-d6c8feca2f0d.png)